COMPILING.md

```diff
@@ -15,10 +15,106 @@ past, but are not actively tested:
 - FreeBSD 11
 - Cygwin
 
-The rest of this document is split up into three parts: compilation on Linux,
-MacOS, Windows.  Please read the section appropriate for your machine.
+# Building using CMAKE
 
-# COMPILATION ON LINUX
+Building with CMake is supported across Linux, MacOS X and Windows with Visual
+Studio. There are also hand-written make files which can be used to build
+separate binaries independently. Usage of the hand-written make files is
+explained in a separate section. The CMake build can build the complete
+repository in fewer steps and supports better integration with various IDEs and
+static-analysis tools. On Windows, the CMake build has the advantage of not
+depending on Cygwin or MinGW, and doesn't require manual modification of build
+files.
+
+1. Ensure you have all the build dependencies installed. Build dependencies are
+   the same as for the makefile build, but with the addition of CMake version
+   3.2 or higher. The installed CMake version can be queried with `cmake
+   --version`. To install cmake:
+   - On Debian-like distributions, do
+     ```
+     apt-get install g++ gcc flex bison make git curl patch cmake
+     ```
+   - On Red Hat/Fedora or derivates, do
+     ```
+     dnf install gcc gcc-c++ flex bison curl patch cmake
+     ```
+   - On macOS, do
+     ```
+     xcode-select --install
+     ```
+     You shoud also install [Homebrew](https://brew.sh), after which you can
+     run `brew install cmake` to install CMake.
```

Comment:
Also similar to tautshnig's comment below, why is MacOS different in style and arguments to the others. I think it would be better to have single (displayed) comment for each system and let that do it all. If users want to play around, this can (probably) be covered elsewhere. (Note, I'm reviewing by commit, so they may be a little haphazard depending on the commits.)